packages/venia-concept/src/components/ForgotPassword/forgotPassword.js

```diff
@@ -0,0 +1,71 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+
+import classify from 'src/classify';
+import ForgotPasswordForm from './ForgotPasswordForm';
+import FormSubmissionSuccessful from './FormSubmissionSuccessful';
+import defaultClasses from './forgotPassword.css';
+import { resetPasswordRequest } from './api';
+
+class ForgotPassword extends Component {
+    static propTypes = {
+        classes: PropTypes.shape({
+            instructions: PropTypes.string
+        }),
+        initialValues: PropTypes.shape({
+            email: PropTypes.string
+        }),
+        onClose: PropTypes.func.isRequired
+    };
+
+    state = {
+        submittedEmail: '',
+        submitSucceeded: false
+    };
+
+    handleFormSubmit = async ({ email }) => {
+        await resetPasswordRequest({ email });
```

Comment:
I'm still very new to the PWA architecture but why are we handling API-level calls within the component? Should it call a function from https://github.com/magento-research/pwa-studio/blob/release/2.0/packages/venia-concept/src/actions/user/asyncActions.js#L1 ? Or like you mentioned, that is out of scope and will be handled with backend?

Reply:
While the actual API call is out of scope, I think you're absolutely right that we should be doing this via Redux actions and not at the component level.

We can dummy the response with a `Promise.resolve()` for now.

@gil-- great catch, thanks!

Reply:
@supernova-at, @gil-- There is no reason to move this API call to user `asyncActions.js` file. We are not going to affect user domain in Redux in all scenarios related to the story. In general, I agree that it is better to store business logic and implementation of complex scenarios in Redux. But here it is just one API call and the page is pretty simple, the only data we need in this story is `isLoading` flag and `error` string for API request. I think we can discuss implementation of separate Redux domain for Forgot Password page, but we definitely do need to add this API call to user asyncActions. In general, in my opinion, it is OK to store data in local state when you do not need to share it. Also, I created a special file `api.js` and implemented a function that makes API call.

Reply:
> I'm still very new to the PWA architecture but why are we handling API-level calls within the component?

Your instinct is correct: we don't and we shouldn't. Since the API call is being mocked here, I don't think it matters. But when the actual API call is implemented, we'll make the call from an async action creator in the `user` domain.

@Starotitorov Please try to follow the patterns in the existing code. These things make it more difficult for us to do code review.

Reply:
@jimbo, but why do we need to put this call in use domain, we are not going to save some data in this domain?

Reply:
What exactly to you want to move to Redux from the changes from this pull request

Reply:
Usually I create a separate domains and add REST API calls to asyncActions, but for this particular case I do not understand why we need it.

Reply:
Real API call will just send an email to the user and that is all.

Reply:
We can consider implementing separate domain for Forgot Password page, but it seems that local state will be enough.

Reply:
@Starotitorov don't worry about it, I created #638 as a follow up.

It's largely a consistency concern on my end.